plugins/gdk-pixbuf/pixbufloader-jxl.c

```diff
@@ -234,19 +473,28 @@ static gboolean load_increment(gpointer context, const guchar *buf, guint size,
         }
         break;
       }
+
       case JXL_DEC_FULL_IMAGE: {
         // TODO(veluca): consider doing partial updates.
         if (decoder_state->area_updated_callback) {
+          GdkPixbuf *output = g_array_index(decoder_state->frames,
+                                            GdkPixbufJxlAnimationFrame, 0)
+                                  .data;
           decoder_state->area_updated_callback(
-              decoder_state->output, 0, 0,
-              gdk_pixbuf_get_width(decoder_state->output),
-              gdk_pixbuf_get_height(decoder_state->output),
-              decoder_state->user_data);
+              output, 0, 0, gdk_pixbuf_get_width(output),
+              gdk_pixbuf_get_height(output), decoder_state->user_data);
         }
-        // TODO(veluca): for animations, this will eventually only leave the
-        // last frame.
+        g_array_index(decoder_state->frames, GdkPixbufJxlAnimationFrame,
+                      decoder_state->frames->len - 1)
+            .decoded = TRUE;
         break;
       }
+
+      case JXL_DEC_SUCCESS: {
+        decoder_state->done = TRUE;
+        return TRUE;
+      }
+
       default: {
```

Comment:
JXL_DEC_ERROR could be handled separately and give a less confusing error message than "Unexpected JxlDecoderProcessInput return code: 1"